openwisp_notifications/api/views.py

```diff
@@ -0,0 +1,56 @@
+import uuid
+
+from openwisp_notifications.api.serializers import NotificationSerializer
+from openwisp_notifications.swapper import load_model
+from rest_framework import status
+from rest_framework.generics import GenericAPIView
+from rest_framework.mixins import ListModelMixin
+from rest_framework.pagination import PageNumberPagination
+from rest_framework.permissions import DjangoModelPermissions
+from rest_framework.response import Response
+
+Notification = load_model('Notification')
+
+
+class NotificationPaginator(PageNumberPagination):
+    page_size = 10
+
+
+class BaseNotificationView(GenericAPIView):
+    model = Notification
+    permission_classes = [DjangoModelPermissions]
+    queryset = Notification.objects.all()
+    serializer_class = NotificationSerializer
+
+    def get_queryset(self):
+        user = self.request.user
+        return self.queryset.filter(recipient=user)
+
+
+class NotificationListView(BaseNotificationView, ListModelMixin):
+    pagination_class = NotificationPaginator
+
+    def get(self, request, *args, **kwargs):
+        return self.list(request, *args, **kwargs)
+
+
+class NotificationReadView(BaseNotificationView, UpdateModelMixin):
+    def patch(self, request, notification_id):
+        return self._mark_notification_read(notification_id)
+
+    def _mark_notification_read(self, notification_id):
+        queryset = self.get_queryset()
+        try:
+            notification_id = uuid.UUID(notification_id)
+            notification = queryset.get(id=notification_id)
+        except ValueError:
+            return Response(status=status.HTTP_400_BAD_REQUEST)
```

Comment:
@pandafy would not be nice for us to give users some more information as to why their response didn't go through? It this case; it was caused by invalid `notification_id`